bftengine/src/bcstatetransfer/DBDataStore.cpp

```diff
@@ -367,28 +381,33 @@ void DBDataStore::deleteAllPendingPages() {
   inmem_->deleteAllPendingPages();
 }
 
-void DBDataStore::deleteCoveredResPageInSmallerCheckpointsTxn( uint64_t minChkp, ITransaction* txn) {
+void DBDataStore::deleteCoveredResPageInSmallerCheckpointsTxn( uint64_t minChkp, ITransaction* txn){
   LOG_DEBUG(logger()," min chkp: " << minChkp << " txn: " << txn->getId());
   auto pages = inmem_->getPagesMap();
-
-  for (auto it = std::find_if(pages.begin(),
-                              pages.end(),
-                              [minChkp](const std::pair<InMemoryDataStore::ResPageKey,
-                                                        InMemoryDataStore::ResPageVal>& elt) {
-                                                       return elt.first.checkpoint < minChkp;});
-        it != pages.end();
-        it++){
-    txn->del(staticResPageKey (it->first.pageId, it->first.checkpoint));
```

Comment:
Do we no longer want to delete static reserved pages?

Reply:
I found the prior version of this code much easier to read. Is there any way to make it look more like that? If not, can you please leave a comment describing the goal for the loop?

Reply:
txn->del(staticResPageKey (it->first.pageId, it->first.checkpoint)); 
was deleting only a static_key -> dynamic_key mapping, not the page itself

Reply:
The previous code was wrong - now I do exactly what is done in InMemoryDB